From stable-bounces@linux.kernel.org  Wed Jul  5 13:48:03 2006
Date: Wed, 5 Jul 2006 22:47:06 +0200
From: Willy Tarreau <w@1wt.eu>
To: marcelo@kvack.org, davem@davemloft.net
Cc: stable@kernel.org, linux-kernel@vger.kernel.org, matthew@wil.cx
Subject: [stable] [PATCH-2.4] 2 oopses in ethtool

From: Willy Tarreau <willy@wtap.(none)>
Date: Wed, 5 Jul 2006 22:34:52 +0200
Subject: [PATCH] ethtool: oops in ethtool_set_pauseparam()

The function pointers which were checked were for their get_* counterparts.
Typically a copy-paste typo.

Signed-off-by: Willy Tarreau <w@1wt.eu>
Acked-by: Jeff Garzik <jeff@garzik.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 net/core/ethtool.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.17.3.orig/net/core/ethtool.c
+++ linux-2.6.17.3/net/core/ethtool.c
@@ -437,7 +437,7 @@ static int ethtool_set_pauseparam(struct
 {
 	struct ethtool_pauseparam pauseparam;
 
-	if (!dev->ethtool_ops->get_pauseparam)
+	if (!dev->ethtool_ops->set_pauseparam)
 		return -EOPNOTSUPP;
 
 	if (copy_from_user(&pauseparam, useraddr, sizeof(pauseparam)))
